src/main/java/pl/wojciechkostecki/revel/security/JwtTokenFilter.java

```diff
@@ -0,0 +1,63 @@
+package pl.wojciechkostecki.revel.security;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+
+@Component
+public class JwtTokenFilter extends OncePerRequestFilter {
```

Comment:
You have to understand how this filter works and what filter is in spring security if you will have this code in your project.